mmap+memset+fork.c

```diff
@@ -106,6 +106,7 @@ int main(int argc, char *argv[]){
 	array = malloc (sizeof(*array)*(i_till-i_from));
 	
 	if (!array) {
+		free(array);
```

Comment:
Don't need to free here as we know that `array` wasn't allocated (since its in the `!array` condition).